### ISSUE TITLE
LSP Sample ?

### ISSUE THREAD

**Issue by melMass** [vshaxe/vshaxe#369]
Hi @Gama11, thanks a lot for all the work on VSHaxe!

I'm implementing a Language Server for a custom language and I would love to use Haxe to do so... 
I've been trying to figure it out from this repo but it's quite hard to isolate the bare minimum for my use case.

It seems I need at least your LSP Haxe implementation, your JsonRPC Haxe implementation, is that right ?

Could you point me to the most simple example of using the tools you've made for vshaxe to create a language server like the official LSP Sample ?

Thanks a lot

**Comment by Gama11** [comment]
I don't think we currently have a hello-world sample for a language server written in Haxe, but that would be nice to have... (similar to how we have a minimal sample [for the VSCode externs](https://github.com/vshaxe/vscode-extern/tree/master/example))

It basically comes down to using https://github.com/vshaxe/language-server-protocol-haxe (the type definitions for the language server protocol) as a library. And that indeed has a dependency on https://github.com/vshaxe/vscode-json-rpc. The simplest option would be to target nodejs, then you can use the Node-specific APIs that haxe-language-server also uses (for instance `MessageReader` / `MessageWriter` [for setting up the communication](https://github.com/vshaxe/haxe-language-server/blob/master/src/haxeLanguageServer/Main.hx)).

If you then want to use the language server in a VSCode extension, you'd have to use the [vscode-languageclient](https://www.npmjs.com/package/vscode-languageclient) library. [We do have externs for that](https://github.com/vshaxe/vshaxe/blob/master/src/vshaxe/server/LanguageClient.hx), but they're not extracted into a separate library. This is used to [start the language server](https://github.com/vshaxe/vshaxe/blob/f25af2696ed02220dcc2986d4d05744a91e493cb/src/vshaxe/server/LanguageServer.hx#L163-L189).

**Comment by melMass** [comment]
Thanks for the reply 
I will try to do a basic sample based on your inputs once I'm done with the TypeScript version.
The issue I had while trying to assemble a simple sample, was that some definitions are scattered across repos (TextDocument for instance) but I will try harder

**Comment by Gama11** [comment]
Hm.. not quite sure what you mean by that.
